airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresJdbcSourceAcceptanceTest.java

```diff
@@ -523,4 +781,139 @@ public void testUserHasNoPermissionToDataBase() throws Exception {
+            .withStreamNamespace(airbyteStream.getStream().getNamespace())
+            .withCursorField(List.of(cursorField))
+            .withCursor(endCursorValue)
+            .withCursorRecordCount(1L));
+
+    final List<AirbyteMessage> expectedMessages = new ArrayList<>(expectedRecordMessages);
+    expectedMessages.addAll(createExpectedTestMessages(expectedStreams));
+
+    assertEquals(expectedMessages.size(), actualMessages.size());
+    assertTrue(expectedMessages.containsAll(actualMessages));
+    assertTrue(actualMessages.containsAll(expectedMessages));
+  }
+
+  @Override
+  protected JsonNode getStateData(final AirbyteMessage airbyteMessage, final String streamName) {
+    final JsonNode streamState = airbyteMessage.getState().getStream().getStreamState();
+    if (streamState.get("stream_name").asText().equals(streamName)) {
+      return streamState;
+    }
+
+    throw new IllegalArgumentException("Stream not found in state message: " + streamName);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param syncState
+   */
+  @Override
+  protected void addStandardStateTypeToSyncState(final JsonNode syncState) {
+    ((ObjectNode) syncState).put("state_type", "standard");
+  }
+
+  /**
+   * {ine}
+   *
+   * @param configuredCatalog catalog of DB source
+   * @return
+   * @throws Exception
+   */
+  @Override
+  protected List<AirbyteMessage> readCatalogForInitialSync(final ConfiguredAirbyteCatalog configuredCatalog) throws Exception {
+    return MoreIterators
+        .toList(source.read(config, configuredCatalog, null));
+  }
+
+  /*
+   * Similar to parent's method but does not include the Data key as CTID, Xmin, and Standard sync no
+   * longer emits this information
+   */
+  @Override
+  protected AirbyteMessage createStateMessage(final DbStreamState dbStreamState, final List<DbStreamState> legacyStates) {
+    final CtidFeatureFlags ctidFeatureFlags = new CtidFeatureFlags(config);
+    if (!ctidFeatureFlags.isCursorSyncEnabled()) {
+      return super.createStateMessage(dbStreamState, legacyStates);
+    }
+
+    if (supportsPerStream()) {
+      return new AirbyteMessage().withType(Type.STATE)
+          .withState(
+              new AirbyteStateMessage().withType(AirbyteStateType.STREAM)
+                  .withStream(new AirbyteStreamState()
+                      .withStreamDescriptor(new StreamDescriptor().withNamespace(dbStreamState.getStreamNamespace())
+                          .withName(dbStreamState.getStreamName()))
+                      .withStreamState(Jsons.jsonNode(dbStreamState))));
+    } else {
+      return new AirbyteMessage().withType(Type.STATE).withState(new AirbyteStateMessage().withType(AirbyteStateType.LEGACY));
+    }
+  }
+
+  /*
+   * Identical to parent method but have a state_type attach to the DBStreamState making it
+   * effectively StandardStatus
+   */
+  @Override
+  protected DbStreamState buildExpectedStreamState(final String streamName,
```

Comment:
Can we revisit some of these overrides with the addtiion of the new StateManager? Perhaps this is more of a reason to do the refactor we spoke about (adding the version/state_type) to the cursormanager itself so that only one place needs to be updated. WDYT since you've dealt with this more?